### PR TITLE
OFS-258: solving problem with filtering show_historical

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/OFS-258

### CODE REVIEW

File: core/gql_queries.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def get_queryset(cls, queryset, info):
         elif info.context.user.is_superuser:
             return queryset
         else:
-            queryset = queryset.filter(*filter_validity())
+            pass
```
Comment (delcroip) [inline core/gql_queries.py:48]: did you have that *filter_validity() set elsewhere ? if not please add a case where the historical is wished but for other cases *filter_validity() must still be used  (in info ?)

**Review by delcroip — CHANGES_REQUESTED** [review]

File: core/gql_queries.py (hunk 1 above)
Comment (sniedzielski) [reply core/gql_queries.py:48]: Yes. It was in schema.py 
`def resolve_role(self, info, **kwargs):
        if not info.context.user.has_perms(CoreConfig.gql_query_roles_perms):
            raise PermissionError("Unauthorized")
        filters = []
        query = Role.objects

        client_mutation_id = kwargs.get("client_mutation_id", None)
        if client_mutation_id:
            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))

        show_history = kwargs.get('show_history', False)
        if not show_history and not kwargs.get('uuid', None):
            filters += filter_validity(**kwargs)

        is_system_role = kwargs.get('is_system', None)
        # check if we can use default filter validity
        if is_system_role is not None:
            if is_system_role:
                query = query.filter(
                    is_system__gte=1
                )
            else:
                query = query.filter(
                    is_system=0
                )
        return gql_optimizer.query(query.filter(*filters), info)`

Comment (sniedzielski) [reply core/gql_queries.py:48]: it is exactly here 
        ```
        show_history = kwargs.get('show_history', False)
        if not show_history and not kwargs.get('uuid', None):
            filters += filter_validity(**kwargs)
```

**Review by delcroip — APPROVED** [review]

File: core/gql_queries.py (hunk 1 above)
Comment (delcroip) [reply core/gql_queries.py:48]: Just to be sure, can you confirm that this query is not called else where. I fear that user will get all their role even the removed ones

Comment (delcroip) [reply core/gql_queries.py:48]: Hello @edarchis, do you think this change will have impact on the way django is loading the users rights ?

Comment (sniedzielski) [reply core/gql_queries.py]: @delcroip I've changed my idea - in that commit you can take a look into it https://github.com/openimis/openimis-be-core_py/pull/53/commits/9a99fd97c5344117ad9ac9a589fa7aef56c6ce83

**Comment by sniedzielski** [comment]
@delcroip as I wrote to you and in subcomment above - I updated this PR via this latest commit - https://github.com/openimis/openimis-be-core_py/pull/53/commits/9a99fd97c5344117ad9ac9a589fa7aef56c6ce83

**Comment by delcroip** [comment]
Ok the has_perm adds the filter anyway

https://github.com/openimis/openimis-be-core_py/blob/5be80cbc7fc23cd25f7a76adb9f6eb65c70b502e/core/models.py#L339

**Pull request merged**